### DISCUSSION TITLE
TypeError: Cannot read properties of null (reading 'toString') when trying to read

### DISCUSSION THREAD

**Discussion by ChristianNewton** [derekantrican/GAS-ICS-Sync#452] [Q&A] [answered]
When trying to run the script, I get the following error:
```
TypeError: Cannot read properties of null (reading 'toString')
createEvent	@ Helpers.gs:503
processEvent	@ Helpers.gs:291
(anonymous)	@ Code.gs:220
startSync	@ Code.gs:219
```

I have tried it with other calendars and they don't have the same problem. I assume the error lies in the .ics file, so please let me know what you suggest. I unfortunately can not motify the file but if there is something I can modify in the script to fix the error, that would be great. 

Here is a copy of my Code.gs file: https://pastebin.com/PkLL5ics. Thanks in advance!

**Comment by Lonestarjeepin** [comment]
Is this issue persistent every time this source calendar tries to run or does it run successfully the next time? I'm seeing this same error, but it's very sporadic so I haven't been able to troubleshoot it.

**Reply by ChristianNewton** [reply]
I had it running all day and it didn't sucessfully sync once. I tried with a few different source calendars and they worked immediately. I did try adding it to Google Calendar with the built-in option and it seemed to work fine.

**Comment by jonas0b1011001** [comment, marked as answer]
Seems like the failing event has an alarm without the 'trigger' property set, which is invalid syntax according the ical standard.

1. easy fix
set `var addAlerts = "no"; `

2. option
change helpers.gs
https://github.com/derekantrican/GAS-ICS-Sync/blob/71bb0a678982178325cc69f44b8b81af86e72a55/Helpers.gs#L502-L513
to
```
        for (var valarm of valarms){
          try{
            var trigger = valarm.getFirstPropertyValue('trigger').toString();
            var alarmTime = new ICAL.Time.fromString(trigger);
            trigger = alarmTime.subtractDateTz(icalEvent.startDate).toString();
            if (overrides.length < 5){ //Google supports max 5 reminder-overrides
              var timer = parseNotificationTime(trigger);
              if (0 <= timer && timer <= 40320)
                overrides.push({'method' : 'popup', 'minutes' : timer});
            }
          }catch(e){}
        }
```
This will still fail on said event but not crash the whole script.

**Reply by ChristianNewton** [reply]
Thanks man, those worked pretty well!
